sofa-ark-maven-plugin打包时提示找不到userSettingFile

### Describe the bug

执行`mvn install`打包时，会报错：`execute depenency:tree failed，The specified user settings file does not exist: C:\Users\FlyAbner\.m2\settings.xml`

### Expected behavior

当用户级的settings.xml不存在时, 使用全局配置文件即可。

### Actual behavior

执行命令报错，构建失败

### Steps to reproduce

mvn clean install -Dmaven.javadoc.skip=true -e -X -U

### Environment

- SOFAArk version: 2.1.3
- JVM version (e.g. `java -version`): 1.8.0_73
- OS version (e.g. `uname -a`): Windows 10
- Maven version: 3.6.3
- IDE version:IntelliJ IDEA 2021.2.2 (Ultimate Edition)

详细报错信息：
```
[ERROR] Failed to execute goal com.alipay.sofa:sofa-ark-maven-plugin:2.1.3:repackage (default-cli) on project ecp-start-mem: execute dependency:tree failed -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.alipay.sofa:sofa-ark-maven-plugin:2.1.3:repackage (default-cli) on project ecp-start-mem: execute dependency:tree failed
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:497)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoExecutionException: execute dependency:tree failed
    at com.alipay.sofa.ark.boot.mojo.RepackageMojo.getAllArtifactByMavenTree (RepackageMojo.java:402)
    at com.alipay.sofa.ark.boot.mojo.RepackageMojo.repackage (RepackageMojo.java:331)
    at com.alipay.sofa.ark.boot.mojo.RepackageMojo.execute (RepackageMojo.java:314)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:497)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR]
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :ecp-start-mem
```

--- a/sofa-ark-parent/support/ark-maven-plugin/src/main/java/com/alipay/sofa/ark/boot/mojo/RepackageMojo.java
+++ b/sofa-ark-parent/support/ark-maven-plugin/src/main/java/com/alipay/sofa/ark/boot/mojo/RepackageMojo.java
@@ -392,8 +392,7 @@ public class RepackageMojo extends TreeMojo {
         request.setGoals(goals);
         request.setBatchMode(mavenSession.getSettings().getInteractiveMode());
         request.setProfiles(mavenSession.getSettings().getActiveProfiles());
-        request.setUserSettingsFile(mavenSession.getRequest().getUserSettingsFile());
-        request.setGlobalSettingsFile(mavenSession.getRequest().getGlobalSettingsFile());
+        setSettingsLocation(request);
         Invoker invoker = new DefaultInvoker();
         try {
             InvocationResult result = invoker.execute(request);
@@ -412,6 +411,17 @@ public class RepackageMojo extends TreeMojo {
             if (outputFile.exists()) {
                 outputFile.delete();
             }
+        }
+    }
+
+    private void setSettingsLocation(InvocationRequest request) {
+        File userSettingsFile = mavenSession.getRequest().getUserSettingsFile();
+        if (userSettingsFile != null && userSettingsFile.exists()) {
+            request.setUserSettingsFile(userSettingsFile);
+        }
+        File globalSettingsFile = mavenSession.getRequest().getGlobalSettingsFile();
+        if (globalSettingsFile != null && globalSettingsFile.exists()) {
+            request.setGlobalSettingsFile(globalSettingsFile);
         }
     }
 

--- a/sofa-ark-parent/support/ark-maven-plugin/src/test/java/com/alipay/sofa/ark/boot/mojo/RepackageMojoTest.java
+++ b/sofa-ark-parent/support/ark-maven-plugin/src/test/java/com/alipay/sofa/ark/boot/mojo/RepackageMojoTest.java
@@ -16,17 +16,26 @@
  */
 package com.alipay.sofa.ark.boot.mojo;
 
+import com.google.common.io.Files;
 import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.io.FileUtils;
 import org.apache.maven.artifact.DefaultArtifact;
+import org.apache.maven.execution.DefaultMavenExecutionRequest;
+import org.apache.maven.execution.MavenExecutionRequest;
+import org.apache.maven.execution.MavenSession;
 import org.apache.maven.shared.dependency.graph.DependencyNode;
 import org.apache.maven.shared.dependency.graph.internal.DefaultDependencyNode;
+import org.apache.maven.shared.invoker.DefaultInvocationRequest;
+import org.apache.maven.shared.invoker.InvocationRequest;
 import org.junit.Assert;
 import org.junit.Test;
 
+import java.io.File;
 import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
 import java.net.URL;
+import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.LinkedHashSet;
 import java.util.Set;
@@ -127,4 +136,51 @@ public class RepackageMojoTest {
         }
         return dependencyNode;
     }
+
+    @Test
+    public void testSetSettingsLocation() throws Exception {
+        String userSettingsFilePath = System.getProperty("user.home") + File.separator
+                                      + "user-settings-test.xml";
+        String globalSettingsFilePath = System.getProperty("user.home") + File.separator
+                                        + "global-settings-test.xml";
+        File userSettingsFile = new File(userSettingsFilePath);
+        File globalSettingsFile = new File(globalSettingsFilePath);
+        InvocationRequest request = new DefaultInvocationRequest();
+        invokeSetSettingsLocation(request, userSettingsFilePath, globalSettingsFilePath);
+        Assert.assertNull(request.getUserSettingsFile());
+        Assert.assertNull(request.getGlobalSettingsFile());
+
+        Files.touch(globalSettingsFile);
+        invokeSetSettingsLocation(request, userSettingsFilePath, globalSettingsFilePath);
+        Assert.assertNull(request.getUserSettingsFile());
+        Assert.assertNotNull(request.getGlobalSettingsFile());
+
+        Files.touch(userSettingsFile);
+        invokeSetSettingsLocation(request, userSettingsFilePath, globalSettingsFilePath);
+        Assert.assertNotNull(request.getUserSettingsFile());
+        Assert.assertNotNull(request.getGlobalSettingsFile());
+
+        FileUtils.deleteQuietly(userSettingsFile);
+        FileUtils.deleteQuietly(globalSettingsFile);
+    }
+
+    private void invokeSetSettingsLocation(InvocationRequest request, String userSettingsFilePath,
+                                           String globalSettingsFilePath) throws Exception {
+        RepackageMojo repackageMojo = new RepackageMojo();
+        MavenExecutionRequest executionRequest = new DefaultMavenExecutionRequest();
+        executionRequest.setUserSettingsFile(new File(userSettingsFilePath));
+        executionRequest.setGlobalSettingsFile(new File(globalSettingsFilePath));
+        // 构造对象
+        MavenSession mavenSession = new MavenSession(null, executionRequest, null,
+            new ArrayList<>());
+        Field mavenSessionField = repackageMojo.getClass().getDeclaredField("mavenSession");
+        mavenSessionField.setAccessible(true);
+        mavenSessionField.set(repackageMojo, mavenSession);
+
+        Method setSettingsLocation = repackageMojo.getClass().getDeclaredMethod(
+            "setSettingsLocation", InvocationRequest.class);
+        setSettingsLocation.setAccessible(true);
+        setSettingsLocation.invoke(repackageMojo, request);
+    }
+
 }

